db/src/id.rs

```diff
@@ -136,38 +134,34 @@ impl<'a> Iterator for IdsIter<'a> {
 }
 
 pub struct LiveSet {
-    filter: CuckooFilter<FnvHasher>,
-    distinct: usize,
+    filter: FnvHashMap<Id, u32>,
 }
 
 impl LiveSet {
     pub fn new() -> Self {
         Self {
-            filter: CuckooFilter::with_capacity(cuckoofilter::DEFAULT_CAPACITY),
-            distinct: 0,
+            filter: Default::default(),
         }
     }
 
     pub fn len(&self) -> usize {
-        self.distinct
+        self.filter.len()
     }
 
     pub fn contains(&self, id: &Id) -> bool {
-        self.filter.contains(id)
+        self.filter.contains_key(id)
     }
 
-    pub fn add(&mut self, id: &Id) -> Result<()> {
-        if !self.filter.contains(id) {
-            self.distinct += 1;
-        }
-        self.filter.add(id)?;
-        Ok(())
+    pub fn add(&mut self, id: &Id) {
+        let count = self.filter.entry(id.clone()).or_default();
+        *count += 1;
```

Comment:
You might want to properly handle the overflow. Panic? Not for this PR, but maybe a todo?

Reply:
Well if it's a `u64` it's impossible to overflow by increments of one

Reply:
count was an u32, no?

Reply:
Ah, you changed it to an u64. Yes, in that case overflowing it is technically possible, but not really....